module-3/lab-intro-to-ml/your-code/main.ipynb

```diff
@@ -20,7 +20,8 @@
     "import numpy as np\n",
```

Comment:
- Using apply to change data type in wrong columns. 
```Python
austin[wrong_type_columns] = austin[wrong_type_columns].apply(lambda col: pd.to_numeric(col, errors='coerce'))
```
- "Next, count the number of rows of austin and missing_values." Use size method instead of len in the following and check the differences between both methods. 
```Python
number_rows_austin = len(df_austin)
number_rows_missing = len(missing_values)
```
- I think your code in this question "As you can see from the output, the majority of missing data is in one column called PrecipitationSumInches. What's the number of missing values in this column in ratio to its total number of rows?" is not correct. 
```Python 
print('Number of missing values for PrecipitationSumInches (%): ',round(100*inter['PrecipitationSumInches'].values[0]/len(austin),2))

#Try the following instead 
missing_values['PrecipitationSumInches'].size / austin['PrecipitationSumInches'].size
```
- I guess this question "What is the largest number of events happened in a single day? Enter your answer in the next cell." should be answer with some kind of query. The key word here is "single day".
- Here "Next, populate the actual values in the dummy columns of  austin_fixed." you can do the following: 
```Python 
for event in event_list:
    austin_fixed[event] = austin_fixed['Events'].str.contains(event).astype(int)
```
- Here "In the cell below, convert the Date column values from string to numeric values using toordinal()." you can do this other way:
```Python
from datetime import datetime as dt
austin_fixed['Date'] = austin_fixed['Date'].apply(lambda date_col: dt.strptime(date_col, '%Y-%m-%d').toordinal())
```
- In this one "Your y should be a subset of austin_fixed containing one column TempAvgF.", you can do this way:
```Python
from datetime import datetime as dt
austin_fixed['Date'] = austin_fixed['Date'].apply(lambda date_col: dt.strptime(date_col, '%Y-%m-%d').toordinal())
```
- I would not use the random_state because each time you run the training of the model you get different results and it could be difficult to check the accuracy of your model each time you make the fine tuning of your params. 
- "svm.SVC(kernel='linear') freezes my PC and don't know why".  Try to build your y as explained above and run the linear SVM again. I think it should work. 
- I would encourage you to calculate both the confusion matrix and the precision score as performance metrics of your classifier.